### ISSUE TITLE
[4.0] webcomponents debug

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#21612]
Normally when joomla debug mode is on we use the non minified assets

But that is not true for "some" of the js in the `system/webcomponents/js` and `vendor/webcomponents/js` folders

Screenshot of global config showing just some are non minified

<img width="389" alt="chrome_2018-08-14_13-58-30" src="https://user-images.githubusercontent.com/1296369/44093031-245d8e78-9fca-11e8-96e8-9ecc48e16be7.png">

**Comment by dgrammatiko** [comment]
@brianteeman check for example rules.php under layouts/joomla/form: 
```js
HTMLHelper::_('webcomponent', 'system/webcomponents/joomla-field-permissions.min.js', ['version' => 'auto', 'relative' => true]);
HTMLHelper::_('webcomponent', 'vendor/joomla-custom-elements/joomla-tab.min.js', ['relative' => true, 'version' => 'auto']);
```
there is a missing flag `'detectDebug' => true`

**Comment by brianteeman** [comment]
~Thanks for the pointer. I can fix that on the train to the presentation~

**Comment by brianteeman** [comment]
or maybe not

A quick search shows these three variations - is there some logic  behind the differences?
```

HTMLHelper::_('webcomponent', 'vendor/joomla-custom-elements/joomla-alert.min.js', ['relative' => true, 'version' => 'auto']);
HTMLHelper::_('webcomponent', 'vendor/joomla-custom-elements/joomla-alert.min.js', ['relative' => true, 'version' => 'auto', 'detectBrowser' => false, 'detectDebug' => false]);
HTMLHelper::_('webcomponent', 'system/webcomponents/joomla-field-media.min.js', ['relative' => true, 'version' => 'auto', 'detectBrowser' => false, 'detectDebug' => true]);

```

**Comment by dgrammatiko** [comment]
Tha last one should be the norm...

**Comment by Bakual** [comment]
Would it make sense to add those as default values to the HtmlHelper method? If not needed it could be explicitely disabled instead of us having to explicitely add it everywhere.

**Comment by brianteeman** [comment]
Would save a massive search and replace and ofc no guarantee it wouldn't happen again

**Comment by Bakual** [comment]
On a fast glance there is actually a default defineed. It's just `false`. If we change that to `true` we could save quite a bit of code. Also the `relative` and `detectBrowser` would already be covered.
See https://github.com/joomla/joomla-cms/blob/1daaaac07555fe031d3395638eb7dbc28f648842/libraries/src/HTML/HTMLHelper.php#L843-L845

**Comment by C-Lodder** [comment]
Is detectDebug the boolean that will ensure the uncompressed version of the file is loaded when debug mode is enabled?

**Comment by brianteeman** [comment]
yes

**Comment by C-Lodder** [comment]
Why does that need to be a function arguement?

**Comment by dgrammatiko** [comment]
it's there for B/C

**Comment by mbabker** [comment]
It has use in some edge cases.  I have extensions where I'm pulling a vendor CSS or JS file and not including both a minified and unminified version of it, so I can tell Joomla to save a few CPU cycles and skip deciding if debug mode is enabled and trying to scan the filesystem for variants that will never exist (same for the detectBrowser option).

**Comment by Bakual** [comment]
> it's there for B/C

B/C for a method which is completely new in 4.0.0? Probably not. 

In the other similar methods for scripts and stylesheets, `detectDebug` already defauls to true, so I'd say it's save to change that behavior for the new method so it follows the other behaviors.

See https://github.com/joomla/joomla-cms/pull/21617

**Comment by dgrammatiko** [comment]
@Bakual if you are about to make this change for the webcomponents why don't you also change it for script and stylesheet, eg make the relative true, the version auto, the detectBrowser false and the detectDebug true as default for all of them?
That will save as a lot of repeated flags and still allow users to use the other settings if they want to.

Sensible defaults make sense to me

**Comment by Bakual** [comment]
From what I saw, `detectBrowser` defaults already to `false` in all those methods.
`version` defaults to empty (false) in all cases. So that's consistent as well.
The only thing different is the `relative` one, and that one I didn't change in the other methods because it would be a B/C break for those existing methods.

I wanted to keep my PR as a no-brainer, changing one simple thing which doesn't break anything 😄

**Comment by brianteeman** [comment]
closed see #22158
